client/src/components/inputs/tagSelector.js

```diff
@@ -0,0 +1,31 @@
+import Select from 'react-select';
+import tagsAPI from '../../api/tagsAPI';
+import {useQuery} from '@apollo/client';
+
+const TagSelector = () => {
+
+  const {data, loading} = useQuery(tagsAPI.getTags);
+
+  if(loading) return <p>Loading...</p>
+  
+  console.log(data.getAllTag);
+
+  const options = data.getAllTag.map((tag) => ({
+    value: tag.id,
+    label: tag.name,
+    color: tag.color
+  }));
+  
```

Comment:
Aquí haría falta añadir los colores a las tags.
Yo propondría una de estas dos opciones (o quizás se podría añadir una tercera que fusionase cosa de ambas opciones). Esta decisión se pasará al equipo de diseño para que ellos decidan cuál creen que es la mejor opción para ello.

- Opción 1:

  const colourStyles = {
    control: (styles) => ({ ...styles, backgroundColor: 'white' }),
    multiValue: (styles, { data }) => {
      return {
        ...styles,
        backgroundColor: data.color,
        paddingTop: "5px;",
        paddingBottom: "5px;",
        paddingLeft: "10px;",
        paddingRight: "10px;",
        borderRadius: "25px;",
        minWidth: "100px;",
        alignItems: "center;",
        justifyContent: "center;"
      };
    },
    multiValueLabel: (styles, { data }) => ({
      ...styles,
      fontWeight: "bold;",
      color: "#FFFFFF;",
      letterSpacing: "1px;",
      fontSize: "15px;",
      textAlign: "center;"
    }),
    multiValueRemove: (styles, { data }) => ({
      ...styles,
      backgroundColor: data.color,
      color: 'white',
      cursor: "pointer",
      ':hover': {
        backgroundColor: data.color,
        color: 'white',
        cursor: "pointer",
      },
    }),
  };


- Opción 2:

  const colourStyles = {
    control: (styles) => ({ ...styles, backgroundColor: 'white' }),
    option: (styles, { data, isDisabled, isFocused, isSelected }) => {
      const color = chroma(data.color);
      return {
        ...styles,
        backgroundColor: isDisabled
          ? undefined
          : isSelected
          ? data.color
          : isFocused
          ? color.alpha(0.1).css()
          : undefined,
        color: isDisabled
          ? '#ccc'
          : isSelected
          ? chroma.contrast(color, 'white') > 2
            ? 'white'
            : 'black'
          : data.color,
        cursor: isDisabled ? 'not-allowed' : 'default',
  
        ':active': {
          ...styles[':active'],
          backgroundColor: !isDisabled
            ? isSelected
              ? data.color
              : color.alpha(0.3).css()
            : undefined,
        },
      };
    },
    multiValue: (styles, { data }) => {
      const color = chroma(data.color);
      return {
        ...styles,
        backgroundColor: color.alpha(0.1).css(),
      };
    },
    multiValueLabel: (styles, { data }) => ({
      ...styles,
      color: data.color,
    }),
    multiValueRemove: (styles, { data }) => ({
      ...styles,
      color: data.color,
      ':hover': {
        backgroundColor: data.color,
        color: 'white',
      },
    }),
  };